ocaml/sdk-gen/component-test/spec/go_test.json

```diff
@@ -0,0 +1,877 @@
+{
+  "test_id1": {
+    "method": [
+      "session.login_with_password",
+      "pool.get_all",
+      "pool.get_record",
+      "host.get_record",
+      "session.logout"
+    ],
+    "params": {
+      "session.login_with_password": ["", "", "1.0", "Go SDK component test"],
+      "pool.get_all":  ["OpaqueRef:12345678-4d8a-e4a9-a241-1ea05c1d6707"],
+      "pool.get_record": ["OpaqueRef:12345678-4d8a-e4a9-a241-1ea05c1d6707", "OpaqueRef:9c81868a-4d8a-e4a9-a241-1ea05c1d6707"],
+      "host.get_record": ["OpaqueRef:12345678-4d8a-e4a9-a241-1ea05c1d6707", "OpaqueRef:4a0fde9c-5709-4f66-9968-64fb6162bd97"],
+      "session.logout": ["OpaqueRef:12345678-4d8a-e4a9-a241-1ea05c1d6707"]
+    },
+    "expected_result": {
+      "session.login_with_password": {
+        "result": "OpaqueRef:12345678-4d8a-e4a9-a241-1ea05c1d6707"
+      },
+      "pool.get_all": {
+        "result": ["OpaqueRef:9c81868a-4d8a-e4a9-a241-1ea05c1d6707"]
+      },
+      "pool.get_record": {
+        "result": {
+          "master" :"OpaqueRef:4a0fde9c-5709-4f66-9968-64fb6162bd97"
+        }
+      },
+      "host.get_record": {
+        "result": {
+          "name_label": "host0",
+          "software_version": {
+            "xapi": "24.11"
+          },
+          "API_version_minor": "21",
+          "API_version_major": "2"
+        }
+      },
+      "session.logout": {
+        "result": ""
+      }
+    }
+  },
+  "test_id2": {
+    "method": [
+      "session.login_with_password", 
+      "pool.get_all", 
+      "pool.get_record",
+      "host.get_record",
+      "VM.get_all_records",
+      "session.logout"
+    ],
+    "params": {
+      "session.login_with_password": ["", "", "1.0", "Go SDK component test"],
+      "pool.get_all":  ["OpaqueRef:12345678-4d8a-e4a9-a241-1ea05c1d6707"],
+      "pool.get_record": ["OpaqueRef:12345678-4d8a-e4a9-a241-1ea05c1d6707", "OpaqueRef:9c81868a-4d8a-e4a9-a241-1ea05c1d6707"],
+      "host.get_record": ["OpaqueRef:12345678-4d8a-e4a9-a241-1ea05c1d6707", "OpaqueRef:4a0fde9c-5709-4f66-9968-64fb6162bd97"],
+      "VM.get_all_records": ["OpaqueRef:12345678-4d8a-e4a9-a241-1ea05c1d6707"],
+      "session.logout": ["OpaqueRef:12345678-4d8a-e4a9-a241-1ea05c1d6707"]
+    },
+    "expected_result": {
+      "session.login_with_password": {
+        "result": "OpaqueRef:12345678-4d8a-e4a9-a241-1ea05c1d6707"
+      },
+      "pool.get_all": {
+        "result": ["OpaqueRef:9c81868a-4d8a-e4a9-a241-1ea05c1d6707"]
+      },
+      "pool.get_record": {
+        "result": {
+          "master" :"OpaqueRef:4a0fde9c-5709-4f66-9968-64fb6162bd97"
+        }
+      },
+      "host.get_record": {
+        "result": {
+          "name_label": "host0",
+          "software_version": {
+            "xapi": "24.11"
+          },
+          "API_version_minor": "21",
+          "API_version_major": "2"
+        }
+      }, 
+      "VM.get_all_records": {
+        "result": {
```

Comment:
If there is a new added field for VM in the future e.g. `groups`, will it affect the test result?

Reply:
No, it will not affect the results, the component test is for testing the method serialize and deserialize, Only need to cover different types.